### PR TITLE
Core/Opcodes:

### PR DESCRIPTION
**Pull request by Golrag**
- SMSG_POWER_UPDATE
- CMSG_SET_SHEATHED
- SMSG_ATTACKSTOP

Any feedback is welcome ! :)
And... About ObjectGuid, how many bytes does it take ? 8 or 16 (talking about size in ServerPacket)

### CODE REVIEW

**Comment by DDuarte** [comment]
ObjectGuid takes 16 bytes in memory and usually less than 16 in packets. It's safe to use 16 when calculating the estimated packet size.

**Comment by Golrag** [comment]
Updated & thank you for the information :)

File: src/server/game/Entities/Unit/Unit.cpp [hunk 1]
```diff
@@ -11465,12 +11464,19 @@ void Unit::SetPower(Powers power, int32 val)
...
     if (IsInWorld())
     {
-        WorldPacket data(SMSG_POWER_UPDATE, 8 + 4 + 1 + 4);
-        data << GetPackGUID();
-        data << uint32(1); //power count
-        data << uint8(powerIndex);
-        data << int32(val);
-        SendMessageToSet(&data, GetTypeId() == TYPEID_PLAYER);
+        WorldPackets::Combat::PowerUpdate packet;
+        WorldPackets::Combat::PowerUpdatePower _power;
```
Comment (Intel) [inline src/server/game/Entities/Unit/Unit.cpp]: We use _ prefix only for private class members :)

Comment (Golrag) [inline src/server/game/Entities/Unit/Unit.cpp]: Updated !

**Pull request merged**